DonoGear/app/src/main/java/com/example/donogear/actionpages/MainActivity.java

```diff
@@ -197,6 +210,52 @@ private void readData() {
                 Log.e("Error", e.toString());
             }
         });
+
+        ParseQuery<ParseObject> announcementQuery = ParseQuery.getQuery(ANNOUNCEMENTS);
+        announcementQuery.findInBackground((announcements, e) -> {
+
+            if (e == null) {
+                for (ParseObject announcement: announcements) {
+                    String announcementId = announcement.getObjectId();
+                    final List<File> announcementImageList = getImageForAnnouncement(announcementId);
+                    System.out.println(announcementImageList.size());
+                    final String announcementTitle = announcement.getString("title");
+                    System.out.println(announcementTitle);
+                    final String announcementDescription = announcement.getString("description");
+                    AnnouncementDetails announcementObject = new AnnouncementDetails(announcementId,
+                            announcementTitle, announcementDescription, announcementImageList);
+                    announcementDetailsList.add(announcementObject);
+                    announcementAdapter.notifyDataSetChanged();
+                }
+            } else {
+                // Something is wrong
+                Toast.makeText(MainActivity.this, "Error: " + e, Toast.LENGTH_SHORT).show();
+                Log.e("Error", e.toString());
+            }
+        });
+    }
+
+
+    private List<File> getImageForAnnouncement(String announcementId) {
+        ParseQuery<ParseObject> announcementImageQuery = ParseQuery.getQuery(ANNOUNCEMENTS);
+        List<File> imageFileList = new ArrayList<>();
+        announcementImageQuery.whereEqualTo("objectId", announcementId);
+        announcementImageQuery.getFirstInBackground((object, e) -> {
+            if (e == null) {
+                if (object.getParseFile("image") != null) {
+                    try {
+                        if (object.getParseFile("image").getFile() != null) {
+                            File imageFile = object.getParseFile("image").getFile();
+                            imageFileList.add(imageFile);
+                            announcementAdapter.notifyDataSetChanged();
+                        }
+                    } catch (ParseException ex) {
+                        Log.e("Error", e.toString());
```

Comment:
You should add the exception to the log as well. Use "Log.e("Error", e.toString(), ex)". Its typically good practice to include the entire exception so that the whole stack trace can be recorded